### PR TITLE
SED-2735 improve-support-of-collection-renaming

### PR DESCRIPTION
**Pull request by david-stephan**

### CODE REVIEW

File: step-framework-parent/step-framework-collections/src/main/java/step/core/collections/inmemory/InMemoryCollection.java [hunk 1]
```diff
@@ -44,7 +44,9 @@ public class InMemoryCollection<T> extends AbstractCollection<T> implements Coll
 	private final ObjectMapper mapper = DefaultJacksonMapperProvider.getObjectMapper();
 
 	private boolean byPassCloning;
-	
+	private InMemoryCollectionFactory parentFactory;
+	private String nane;
```
Comment (jeromecomte) [inline step-framework-parent/step-framework-collections/src/main/java/step/core/collections/inmemory/InMemoryCollection.java:48]: Typo

File: step-framework-parent/step-framework-collections/src/main/java/step/core/collections/inmemory/InMemoryCollection.java [hunk 2]
```diff
@@ -192,7 +196,11 @@ public void createOrUpdateCompoundIndex(Map<String, Integer> fields) {
 
 	@Override
 	public void rename(String newName) {
-		// TODO Auto-generated method stub
+		if (parentFactory != null) {
+			parentFactory.renameCollection(nane, newName);
+		} else {
+			throw new UnsupportedOperationException("renaming collection is not supported for in memory collections");
```
Comment (jeromecomte) [inline step-framework-parent/step-framework-collections/src/main/java/step/core/collections/inmemory/InMemoryCollection.java:202]: Not sure if we should throw an exception in this case or simply do nothing as we don't have any parent factory

File: step-framework-parent/step-framework-collections/src/main/java/step/core/collections/inmemory/InMemoryCollection.java [hunk 3]
```diff
@@ -192,7 +196,11 @@ public void createOrUpdateCompoundIndex(Map<String, Integer> fields) {
 
 	@Override
 	public void rename(String newName) {
-		// TODO Auto-generated method stub
+		if (parentFactory != null) {
```
Comment (jeromecomte) [inline step-framework-parent/step-framework-collections/src/main/java/step/core/collections/inmemory/InMemoryCollection.java:199]: I assume we should change the value of the field "name" too

File: step-framework-parent/step-framework-collections/src/test/java/step/core/collections/inmemory/InMemoryCollectionTest.java [hunk 4]
```diff
@@ -9,4 +9,9 @@ public class InMemoryCollectionTest extends AbstractCollectionTest {
     public InMemoryCollectionTest() {
         super(new InMemoryCollectionFactory(new Properties()));
     }
+
+    @Override
+    public void renameCollections() {
+        //not implemented for in memory
```
Comment (jeromecomte) [inline step-framework-parent/step-framework-collections/src/test/java/step/core/collections/inmemory/InMemoryCollectionTest.java:15]: Actually it should work too. Here I would also check that the value of the field "name" has been changed

**Review by jeromecomte — CHANGES_REQUESTED** [review]

File: step-framework-parent/step-framework-collections/src/main/java/step/core/collections/inmemory/InMemoryCollection.java (hunk 3 above)
Comment (david-stephan) [reply step-framework-parent/step-framework-collections/src/main/java/step/core/collections/inmemory/InMemoryCollection.java:199]: Correct, fix in next commit

**Review by jeromecomte — APPROVED** [review]

File: step-framework-parent/step-framework-collections/src/main/java/step/core/collections/inmemory/InMemoryCollection.java [hunk 5]
```diff
@@ -196,10 +196,10 @@ public void createOrUpdateCompoundIndex(Map<String, Integer> fields) {
 
 	@Override
 	public void rename(String newName) {
+		//Renaming only make sense when created from a factory
 		if (parentFactory != null) {
-			parentFactory.renameCollection(nane, newName);
-		} else {
-			throw new UnsupportedOperationException("renaming collection is not supported for in memory collections");
+			parentFactory.renameCollection(name, newName);
+			this.name = newName;
```
[the remark this answers is not in the document]
Comment (david-stephan) [reply step-framework-parent/step-framework-collections/src/main/java/step/core/collections/inmemory/InMemoryCollection.java]: applied this last change.

**Pull request merged**